dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java

```diff
@@ -28,21 +29,24 @@
 
 public class WebProxyServlet extends ProxyServlet {
 
-  private String targetAddress;
-
   private static final Logger LOG = LoggerFactory.getLogger(WebProxyServlet.class);
 
-  public WebProxyServlet(String targetAddress) {
-    this.targetAddress = targetAddress;
+  private Map<String, String> serverAddressesMap;
+
+  public WebProxyServlet(Map<String, String> serverAddressesMap) {
+    this.serverAddressesMap = serverAddressesMap;
   }
 
   @Override
   protected String rewriteTarget(HttpServletRequest clientRequest) {
     if (!validateDestination(clientRequest.getServerName(), clientRequest.getServerPort())) {
       return null;
     }
+    String targetAddress = serverAddressesMap.get(clientRequest.getHeader("targetAddress"));
+    if (targetAddress == null) {
+      targetAddress = "http://localhost:19997/";
```

Comment:
I don't think we should handle this unexpected situation. because the protocol, host and port are not fixed.

Reply:
> I don't think we should handle this unexpected situation. because the protocol, host and port are not fixed.

Have deleted.